vnext/ReactUWP/Modules/Animated/SpringAnimationDriver.cpp

```diff
@@ -0,0 +1,146 @@
+                     .dereference()
+                     .second.asDouble());
+}
+
+std::tuple<winrt::CompositionAnimation, winrt::CompositionScopedBatch>
+SpringAnimationDriver::MakeAnimation(const folly::dynamic &config) {
+  const auto [scopedBatch, animation, easingFunction] = []() {
+    const auto compositor = winrt::Window::Current().Compositor();
+    return std::make_tuple(
+        compositor.CreateScopedBatch(
+            winrt::CompositionBatchTypes::AllAnimations),
+        compositor.CreateScalarKeyFrameAnimation(),
+        compositor.CreateLinearEasingFunction());
+  }();
+
+  const auto startValue = GetAnimatedValue()->RawValue();
+  std::vector<float> keyFrames = [this, startValue]() {
+    std::vector<float> keyFrames;
+    bool done = false;
+    double time = 0;
+    while (!done) {
+      time += 1.0f / 60.0f;
+      auto [currentValue, currentVelocity] =
+          GetValueAndVelocityForTime(time, startValue);
+      keyFrames.push_back(currentValue);
+      if (IsAtRest(currentVelocity, currentValue, startValue) ||
+          (m_overshootClampingEnabled &&
+           IsOvershooting(currentValue, startValue))) {
+        done = true;
+      }
+    }
+    return keyFrames;
+  }();
+
+  std::chrono::milliseconds duration(
+      static_cast<int>(keyFrames.size() / 60.0f * 1000.0f));
+  animation.Duration(duration);
+
+  auto normalizedProgress = 0.0f;
+  animation.InsertKeyFrame(normalizedProgress, static_cast<float>(startValue), easingFunction);
+  for (const auto keyFrame : keyFrames) {
+    normalizedProgress = std::min(normalizedProgress + 1.0f / keyFrames.size(), 1.0f);
+    animation.InsertKeyFrame(normalizedProgress, keyFrame, easingFunction);
+  }
+  animation.IterationCount(static_cast<int32_t>(m_iterations));
+  animation.IterationBehavior(winrt::AnimationIterationBehavior::Count);
+
+  return std::make_tuple(animation, scopedBatch);
+}
+
+std::tuple<float, double> SpringAnimationDriver::GetValueAndVelocityForTime(
+    double time,
+    double startValue) {
+  const auto c = m_springDamping;
+  const auto m = m_springMass;
+  const auto k = m_springStiffness;
+  const auto v0 = -m_initialVelocity;
+
+  const auto zeta = c / (2 * std::sqrt(k * m));
+  const auto omega0 = std::sqrt(k / m);
+  const auto omega1 = omega0 * std::sqrt(1.0 - (zeta * zeta));
+  const auto x0 = m_endValue - startValue;
+
+  if (zeta < 1) {
+    const auto envelope = std::exp(-zeta * omega0 * time);
+    const auto value = static_cast<float>(m_endValue -
+        envelope *
+            ((v0 + zeta * omega0 * x0) / omega1 * std::sin(omega1 * time) +
+             x0 * std::cos(omega1 * time)));
+    const auto velocity = zeta * omega0 * envelope *
+            (std::sin(omega1 * time) * (v0 + zeta * omega0 * x0) / omega1 +
+             x0 * std::cos(omega1 * time)) -
+        envelope *
+            (std::cos(omega1 * time) * (v0 + zeta * omega0 * x0) -
+             omega1 * x0 * std::sin(omega1 * time));
```

Comment:
This formatting looks a little strange. Is it clang though?

Reply:
FYI, you need to run `yarn change` in the root of the repo to generate a changelog for this change.  (That's the failure you are getting in the CI build)